### PR TITLE
detect differing origin and indicate it at auth failure

### PR DESCRIPTION
**Pull request by jot-hub**
## Hey, I just made a Pull Request!

fixes: #3223 

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thank you! I like the approach and it's pretty much good to go. Just a minor nit on the contract between the app and the popup to make sure it's easier to evolve.

File: packages/core-api/src/lib/loginPopup.ts [hunk 1]
```diff
@@ -92,6 +94,12 @@ export function showLoginPopup(options: LoginPopupOptions): Promise<any> {
         return;
       }
       const { data } = event;
+
+      if (data.targetOrigin) {
```
[the remark this answers is not in the document]
Comment (jot-hub) [reply packages/core-api/src/lib/loginPopup.ts]: Done! I chose not to pollute the AuthResult / WebMessageResponse types with this

File: packages/core-api/src/lib/loginPopup.ts [hunk 2]
```diff
@@ -111,7 +119,12 @@ export function showLoginPopup(options: LoginPopupOptions): Promise<any> {
 
     const intervalId = setInterval(() => {
       if (popup.closed) {
-        const error = new Error('Login failed, popup was closed');
+        const errMessage = `Login failed, ${
+          targetOrigin !== window.location.origin
+            ? 'incorrect app origin'
```
Comment (Rugvip) [inline packages/core-api/src/lib/loginPopup.ts:124]: Shouldn't be a problem to show the expected origin here as it wouldn't be sensitive information. Also nothing that a regular user would see in a correctly configured app. Something like `Incorrect app origin, expected ${targetOrigin}`

**Review by Rugvip — APPROVED** [review]
Thanks! 👌

And sorry, realized another another nit on the second readthrough that I think makes this even more valuable 😅

File: packages/core-api/src/lib/loginPopup.ts (hunk 2 above)
Comment (jot-hub) [reply packages/core-api/src/lib/loginPopup.ts]: done as well! Thanks!

**Review by Rugvip — APPROVED** [review]
:shipit:, thanks again!

**Pull request merged**